src/hooks/useStaking.js

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from 'react'
+import { useMounted } from './useMounted'
+import { useOrgApps } from '../providers/OrgApps'
+import { useWallet } from '../providers/Wallet'
+
+export function useStaking() {
+  const mounted = useMounted()
+  const { account } = useWallet()
+  const { agreementApp } = useOrgApps()
+  const [stakeManagement, setStakeManagement] = useState(null)
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    async function getStakingInformation() {
```

Comment:
Let's chat about what to do with this hook :) we need to be subscribing and formatting the data and I think there's enough overlap to have it in the `agreementSubscription` and `useAgreement`. I can help with this!